src/Features/Core/Portable/ExternalAccess/UnitTesting/UnitTestingFeaturesReferencesService.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeLens;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting;
+
+internal interface IUnitTestingCodeLensContext
+{
+    Task<ImmutableArray<ReferenceMethodDescriptor>?> FindReferenceMethodsAsync(
+        Guid projectGuid, string filePath, TextSpan span, DocumentId? sourceGeneratedDocumentId, CancellationToken cancellationToken);
+}
+
+internal interface IUnitTestingFeaturesReferencesServiceCallback
+{
+    Task<TResult> InvokeAsync<TResult>(string targetName, IReadOnlyList<object?> arguments, CancellationToken cancellationToken);
+}
+
+internal static class UnitTestingFeaturesReferencesService
+{
+    internal static async Task<ImmutableArray<(string MethodFullyQualifedName, string MethodOutputFilePath)>> GetCallerMethodsAsync(
+        Guid projectGuid,
+        string filePath,
+        TextSpan span,
+        DocumentId? sourceGeneratedDocumentId,
+        IUnitTestingFeaturesReferencesServiceCallback callback,
+        CancellationToken cancellationToken)
```

Comment:
this api is much lower than the prior Editor level api.  So it doesn't talk about vs codelens concepts.  It just talks about data, and just passes in an abstraction for the call to be made back to VS through the 'callback' parameter which hides all the codelens stuff as well.